### PR TITLE
Logger

### PR DESCRIPTION
**Pull request by slavik-lvovsky**

### CODE REVIEW

File: backend/src/logger/settings.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import * as vscode from "vscode"; // NOSONAR
+
+/**
+ * Note that the values of these configuration properties must match those defined in the package.json
+ */
+export const LOGGING_LEVEL_CONFIG_PROP = "Yeoman UI.loggingLevel";
+export const SOURCE_TRACKING_CONFIG_PROP = "Yeoman UI.sourceLocationTracking";
```
Comment (tomer-epstein) [inline backend/src/logger/settings.ts]: "Yeoman UILogger.loggingLevel"
"Yeoman UI.Logger.sourceLocationTracking"

File: backend/package.json [hunk 2]
```diff
@@ -32,6 +32,33 @@
 				}
 			}
 		],
+		"configuration": {
+			"type": "object",
+			"title": "Yeoman UI",
+			"properties": {
+				"Yeoman UI.loggingLevel": {
```
Comment (tomer-epstein) [inline backend/package.json]: Yeoman UI.Logger.loggingLevel

File: backend/package.json [hunk 3]
```diff
@@ -32,6 +32,33 @@
...
+						"warn",
+						"info",
+						"debug",
+						"trace"
+					],
+					"default": "warn",
+					"description": "The verbosity of logging according to the following order: trace > debug > info > warn > error > fatal > off.",
+					"scope": "resource"
+				},
+				"Yeoman UI.sourceLocationTracking": {
```
Comment (tomer-epstein) [inline backend/package.json]: "Yeoman UI.Logger.sourceLocationTracking"

File: backend/package.json [hunk 4]
```diff
@@ -32,6 +32,33 @@
 				}
```
Comment (tomer-epstein) [inline backend/package.json]: It seems that command yeomanUI.toggleLog is not relevant.

File: backend/src/logger/settings.ts (hunk 1 above)
Comment (slavik-lvovsky) [reply backend/src/logger/settings.ts]: Done

File: backend/package.json (hunk 2 above)
Comment (slavik-lvovsky) [reply backend/package.json]: Done

File: backend/package.json (hunk 3 above)
Comment (slavik-lvovsky) [reply backend/package.json]: Done

File: backend/package.json (hunk 4 above)
Comment (slavik-lvovsky) [reply backend/package.json]: Done

**Pull request closed without merging**